### PR TITLE
fix: table options can't be found in distributed mode

### LINKED ISSUE #1433: Table options can't be found in distributed mode
### What type of bug is this?

Incorrect result

### What subsystems are affected?

Frontend

### What happened?

When implementing `SHOW CREATE` statement #1336 , I found that the table options can't be retrieved from table info, it's stored in the region on datanode.

After discussing with @v0y4g3r , @waynexia , maybe we can provide an RPC to get the table info from the table's first region.

### What operating system did you use?

Mac OS

### Relevant log output and stack trace

_No response_

### How can we reproduce the bug?

`show create table [table]` in distributed mode doesn't display the table options info.

### PR DESCRIPTION
**Pull request by NiwakaDev**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This PR fixes that table options can't be found in distributed mode.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)

close #1433

### CODE REVIEW

**Comment by killme2008** [comment]
Great work! Some conflicts have to be resolved. @NiwakaDev

File: tests/cases/distributed/show/show_create.result [hunk 1]
```diff
@@ -62,7 +64,9 @@ show create table table_without_partition;
 |                         | )                                                       |
 |                         |                                                         |
 |                         | ENGINE=mito                                             |
-|                         |                                                         |
+|                         | WITH(                                                   |
+|                         |   regions = 1                                           |
+|                         | )                                                       |
```
Comment (NiwakaDev) [inline tests/cases/distributed/show/show_create.result:69]: I don't know whether or not `greptimedb` should show table options when a corresponding table doesn't define partitions.

**Comment by NiwakaDev** [comment]
I resolved the conflicts.
@killme2008  PTAL.

File: src/frontend/src/instance/distributed.rs [hunk 2]
```diff
@@ -121,6 +121,10 @@ impl DistInstance {
         info!("Successfully created distributed table '{table_name}' with table id {table_id}");
 
         table_info.ident.table_id = table_id;
+
+        // TODO: Fetch a latest region_numbers from MetaSrv so that FrontCatalogManager can use a latest one.
```
Comment (MichaelScofield) [inline src/frontend/src/instance/distributed.rs:125]: Normally we don't put `regions = N` in `with` for distributed table. Actually the part regarding "regions" in the create distributed table sql might just be changed in the future (see https://github.com/GreptimeTeam/greptimedb/issues/1281). So I think we can ignore the "regions" table options here, and test other table options.

The other table options to test are: "ttl" and "write_buffer_size". See https://docs.greptime.com/reference/sql/create#table-options. Also for create external table, there are more table options: https://docs.greptime.com/reference/sql/create#create-external-table.

All in all, we'd like the output of show create table sql behaves the same in both standalone mode and distributed mode.

File: tests/cases/distributed/show/show_create.result (hunk 1 above)
Comment (killme2008) [reply tests/cases/distributed/show/show_create.result:69]: I think both choices are fine, let's keep the current behaviour.

File: src/common/meta/src/key.rs [hunk 3]
```diff
@@ -215,9 +215,14 @@ impl TableMetadataManager {
     /// The caller MUST ensure it has the exclusive access to `TableNameKey`.
     pub async fn create_table_metadata(
         &self,
-        table_info: RawTableInfo,
+        mut table_info: RawTableInfo,
         region_routes: Vec<RegionRoute>,
     ) -> Result<()> {
+        let mut region_numbers = vec![];
```
Comment (killme2008) [inline src/common/meta/src/key.rs:221]:
```suggestion
        let mut region_numbers = Vec::with_capacity(region_routes.len());
```

Or maybe we can use iterator to construct it:

```rust
let region_numbers = region_routes.iter()
    .map(|region| region.region.id.region_number())
    .collect::<Vec<_>>();
```

**Review by killme2008 — APPROVED** [review]
LGTM

File: src/frontend/src/instance/distributed.rs (hunk 2 above)
Comment (NiwakaDev) [reply src/frontend/src/instance/distributed.rs:125]: >Normally we don't put regions = N in with for distributed table.

Why does `greptimedb` show `regions = ~` in `with` for standalone, like so:

https://github.com/GreptimeTeam/greptimedb/blob/9352649f22b558bf7c44f25a41b301271ea045e5/tests/cases/standalone/show/show_create.result#L33-L34

If we don't need to show `regions = ~`, I think that we can remove it not only for distributed mode but also for standalone mode.

Comment (killme2008) [reply src/frontend/src/instance/distributed.rs:125]: Yes, we can remove it right now and leave a TODO. After the new engine is ready, we can go back to add it back.

Comment (MichaelScofield) [reply src/frontend/src/instance/distributed.rs:125]: I don't think we should add `regions = ~` in the output of show create table sql. In standalone mode, there's only one region. In distributed mode, we don't use `with (regions = ~)` to create multiple regions. @killme2008

Comment (killme2008) [reply src/frontend/src/instance/distributed.rs:125]: The standalone will support multi-regions tables in future.

**Review by MichaelScofield — APPROVED** [review]

File: src/common/meta/src/key.rs (hunk 3 above)
Comment (killme2008) [reply src/common/meta/src/key.rs:221]: @NiwakaDev Please apply this suggestion.

File: src/frontend/src/instance/distributed.rs (hunk 2 above)
Comment (NiwakaDev) [reply src/frontend/src/instance/distributed.rs:125]: @MichaelScofield  
Should I remove this TODO ?

Comment (MichaelScofield) [reply src/frontend/src/instance/distributed.rs:125]: @NiwakaDev yes, please. Regarding "regions = ~" in show create SQL's output, let's reconsider it after region server refactor is done. Thx!

File: src/common/meta/src/key.rs (hunk 3 above)
Comment (NiwakaDev) [reply src/common/meta/src/key.rs]: Done: [f009a71](https://github.com/GreptimeTeam/greptimedb/pull/2209/commits/f009a711961ff79f64c692cf30ea6dd2c6ab10ca)

Please let me know if there are any omissions.

File: src/frontend/src/instance/distributed.rs (hunk 2 above)
Comment (NiwakaDev) [reply src/frontend/src/instance/distributed.rs]: Addressed your review: 5939ce5

**Pull request merged**